### PR TITLE
Material properties

### PR DESCRIPTION
**Pull request by StollToaldo**
Add new concrete properties and stress-strain curves for both materials.

### CODE REVIEW

File: structuraldesign/materials/concrete.py [hunk 1]
```diff
@@ -1,12 +1,126 @@
...
+            n = 2
+        elif fck > 50 and fck <= 90:
+            ec2 = 2 + 0.085 * (fck - 50) ** (0.53)
+            ecu = 2.6 + 35 * ((90 - fck) / 100) ** 4
+            n = 1.4 + 23.4 * ((90 - fck) / 100) ** 4
+        
+        ec2 = ec2 / 1000 #1/1000 to no units
+        ecu = ecu / 1000 #1/1000 to no units
+
+        print ec2, ecu, fck, fcd
```
Comment (StollToaldo) [inline structuraldesign/materials/concrete.py]: forgot to delete this line XD

File: structuraldesign/materials/concrete.py [hunk 2]
```diff
@@ -1,12 +1,126 @@
...
+        self.Ecs = self.Eci*self.alpha_i
+
+        self.xd_lim = self._calc_xd_lim(self.fck)
+
+
     def __repr__(self):
         return '<Concrete C{}>'.format(self.fck)
+
+    def _calc_alpha_e(self, aggregate_material):
+        if aggregate_material == 1: #basalto/diabasio
```
Comment (rodrigorahal) [inline structuraldesign/materials/concrete.py]: maybe we should have a dict mapping `alpha_e_by_aggregate_material`:
```
alpha_e_by_aggregate_material = {
    1: 1.2, #material description
    2: 1.0,
    3: 0.9,
    4: 0.7
}

return alpha_e_by_aggregate_material[aggregate_material]
```

File: structuraldesign/materials/concrete.py [hunk 3]
```diff
@@ -1,12 +1,126 @@
...
+        self.aggregate_material = aggregate_material
 
         self.fcd = fck / GAMA_C
 
+        self.fctm = self._calc_fctm(self.fck)
+
+        self.fctk_inf = self._calc_fctk_inf(self.fctm)
+        self.fctk_sup = self._calc_fctk_inf(self.fctm)
+
+        self.fctd = self.fctk_inf/GAMA_C
```
Comment (rodrigorahal) [inline structuraldesign/materials/concrete.py]: Just a styling tip: we sholud separate symbols and variables/numbers by a space caracter: `self.fctk_inf / GAMA_C` it improves readability.

File: structuraldesign/materials/concrete.py [hunk 4]
```diff
@@ -1,12 +1,126 @@
...
+            alpha_e = 0.9
+        else: #arenito
+            alpha_e = 0.7
+
+        return alpha_e
+
+    def _calc_fctm(self, fck):
+        if self.fck <= 50:
+            fctm = 0.3*fck**(2.0/3)
+        elif fck > 50 and fck <=90:
```
Comment (rodrigorahal) [inline structuraldesign/materials/concrete.py]: you can just write: `elif `50 < fck <= 90:` :)

File: structuraldesign/materials/concrete.py [hunk 5]
```diff
@@ -1,12 +1,126 @@
...
+            fctm = 2.12*math.log(1+0.11*fck)
+        else:
+            # return erro ???
```
Comment (rodrigorahal) [inline structuraldesign/materials/concrete.py]: `raise ValueError("Concrete's fck input value should be under 90MPa")

File: structuraldesign/materials/concrete.py [hunk 6]
```diff
@@ -1,12 +1,126 @@
...
+            pass
+
+        return fctm
+
+    def _calc_fctk_inf(self, fctm):
+        fctk_inf = 0.7*fctm
```
Comment (rodrigorahal) [inline structuraldesign/materials/concrete.py]: just `return 0.7 * fctm`?

File: structuraldesign/materials/concrete.py [hunk 7]
```diff
@@ -1,12 +1,126 @@
...
+
+        return fctk_inf
+
+    def _calc_fctk_sup(self, fctm):
+        fctk_sup = 1.3*fctm
```
Comment (rodrigorahal) [inline structuraldesign/materials/concrete.py]: `return 1.3 * fctm`

File: structuraldesign/materials/concrete.py [hunk 8]
```diff
@@ -1,12 +1,126 @@
...
+
+        return fctk_sup
+
+    def _calc_Eci(self, fck, alpha_e):
+        if fck>=20 and fck<=50:
```
Comment (rodrigorahal) [inline structuraldesign/materials/concrete.py]: again: `if 20 <= fck <= 50`: 
python is great :)

File: structuraldesign/materials/concrete.py [hunk 9]
```diff
@@ -1,12 +1,126 @@
...
+            Eci = alpha_e*5600*math.sqrt(fck)
+        elif fck >50 and fck<=90:
+            Eci = 21500*alpha_e*(fck/10+1.25)**(1.0/3)
+
+        return Eci
+
+    def _calc_alpha_i(self, fck):
+        alpha_i = min(0.8+0.2*fck/80.0, 1)
```
Comment (rodrigorahal) [inline structuraldesign/materials/concrete.py]: just `return min(0.8 + 0.2 * fck / 80.0, 1)`

File: structuraldesign/materials/concrete.py [hunk 10]
```diff
@@ -1,12 +1,126 @@
...
+
+        return xd_lim
+
+    def stress(self, strain):
+        """Stress-Strain Curve
+            strain > 0 = compression
+            strain < 0 = tension
+        """
+
+        fck = self.fck
```
Comment (rodrigorahal) [inline structuraldesign/materials/concrete.py]: You should use `self.fck` and `self.fcd` directly. It's easier to know down the line that those are instance variables

File: structuraldesign/materials/concrete.py [hunk 11]
```diff
@@ -1,12 +1,126 @@
...
+                sigma = 0.85 * fcd
+            else:
+                sigma = 0.85 * fcd * (1 - (1 - strain / ec2) ** n)
+                
+        else: #no tensile strength
+            sigma = 0
+    
+        return sigma
+
+c = Concrete(40)
```
Comment (rodrigorahal) [inline structuraldesign/materials/concrete.py]: You can delete this and write a `unittest` in the `tests\` folder. And check if all the initialization is being done correctly. I can help with that.

**Comment by rodrigorahal** [comment]
Great job @StollToaldo! I wrote a few styling comments and few tips to reduce the number of code lines. We should write a unittest for the `ConcreteInitializationTest` before merging this. I can help with that it's pretty easy and very helpful.

File: structuraldesign/materials/concrete.py (hunk 8 above)
Comment (StollToaldo) [inline structuraldesign/materials/concrete.py]: Awesome. I got this function from VBA, which sucks hard.

File: structuraldesign/materials/concrete.py (hunk 10 above)
Comment (StollToaldo) [inline structuraldesign/materials/concrete.py]: Got it. I did this way cause I thought it was easy to write.

**Pull request merged**